tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5956,308 @@ def get_prod(arr):
+              uniform(low=-((abs(min_float64) / (size * 2))**(1/2)), high=(abs(max_float64) / (size * 2))**(1/2), size=size) * 1j
+
+    f_bl = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_bl(bl), get_norm(bl), rtol=RTOL, atol=ATOL)
+
+    f_integer8 = epyccel(get_norm, language=language)
+    f_integer16 = epyccel(get_norm, language=language)
+    f_integer = epyccel(get_norm, language=language)
+    f_integer32 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_integer8(integer8), get_norm(integer8), rtol=RTOL, atol=ATOL)
+    assert np.isclose(f_integer16(integer16), get_norm(integer16), rtol=RTOL, atol=ATOL)
+    assert np.isclose(f_integer(integer), get_norm(integer), rtol=RTOL, atol=ATOL)
+    assert np.isclose(f_integer32(integer32), get_norm(integer32), rtol=RTOL, atol=ATOL)
+
+    f_integer64 = epyccel(get_norm, language=language)
+    assert np.isclose(f_integer64(integer64), get_norm(integer64), rtol=RTOL, atol=ATOL)
+
+    f_fl = epyccel(get_norm, language=language)
+    f_fl32 = epyccel(get_norm, language=language)
+    f_fl64 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_fl(fl), get_norm(fl), rtol=RTOL, atol=ATOL)
+    assert np.isclose(f_fl32(fl32), get_norm(fl32), rtol=RTOL32, atol=ATOL32)
+    assert np.isclose(f_fl64(fl64), get_norm(fl64), rtol=RTOL, atol=ATOL)
+
+    f_complex64 = epyccel(get_norm, language=language)
+    f_complex128 = epyccel(get_norm, language=language)
+
+    assert np.isclose(f_complex64(cmplx64), get_norm(cmplx64), rtol=RTOL32, atol=ATOL32)
+    assert np.isclose(f_complex128(cmplx128), get_norm(cmplx128), rtol=RTOL, atol=ATOL)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Still under maintenance, See #769"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [pytest.mark.python])
+    )
+)
+
+def test_numpy_norm_array_like_2d(language):
+
+    @types('bool[:,:]')
+    @types('int[:,:]')
+    @types('int8[:,:]')
+    @types('int16[:,:]')
+    @types('int32[:,:]')
+    @types('int64[:,:]')
+    @types('float[:,:]')
+    @types('float32[:,:]')
+    @types('float64[:,:]')
+    @types('complex64[:,:]')
+    @types('complex128[:,:]')
+    def get_norm(arr):
+        from numpy.linalg import norm
+        from numpy import shape
+        a = norm(arr, axis=1)
+        s = shape(a)
+        return len(s), s[0], a[0]
+
+    size = (2, 5)
+
+    bl = randint(0, 1, size=size, dtype= bool)
+
+    integer8 = randint(min_int8, max_int8, size=size, dtype=np.int8)
+    integer16 = randint(min_int16, max_int16, size=size, dtype=np.int16)
+    integer = randint(min_int, max_int, size=size, dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=size, dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=size, dtype=np.int64)
```

Comment:
Are you sure there are no overflow errors for the higher precision integers?

Reply:
yes, i tested it.

Reply:
Prod promotes its arguments to floats so this is not a problem